### PR TITLE
Single weather report view work

### LINKED ISSUE #28: Single weather report view
# Overview
We need a GUI for displaying a single weather report item. It should display all the properties in the `WeatherReportModel`.
In its current form it just displays all properties in `TextViews` in no particular style.

It's a complicated layout, so it doesn't have to match this prototype exactly, but this is the sort of style we are aiming for:
Screen Shot 2021-09-28 at 8 42 35 am

### PR DESCRIPTION
**Pull request by calebwebster**
# Overview

This PR will complete the basic layout of the SingleWeatherReport activity.

The format of the activity is an exact replication of the example user story wireframe and contains placeholder values and images.

## Changes

- Changes to xml file
- Comment out findViewById() calls to avoid errors, as the new xml views have not been assigned IDs yet.
- Add a few new .png image files for placeholders
- New drawable city_name_layout_bg.xml file for the blue background of the LinearLayout that displays the city name and state abbreviation.

### ToDo:

- Data insertion needs to be done.
- Buttons need to be hooked up
- Weather data fields may need to be changed (I'm not sure if the data we have matches the wireframe fields, I literally just copied the wireframe)
- Probably an animation for the data drop-down with the little chevron button next to the "Details" text

_Important_: Free images from [Flaticon](https://www.flaticon.com) were used, and authors need to be attributed:

<div>Icons made by <a href="https://www.flaticon.com/authors/vignesh-oviyan" title="Vignesh Oviyan">Vignesh Oviyan</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a></div>

<div>Icons made by <a href="https://www.flaticon.com/authors/fps-web-agency" title="fps web agency">fps web agency</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a></div>

<div>Icons made by <a href="https://www.flaticon.com/authors/good-ware" title="Good Ware">Good Ware</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a></div>

<div>Icons made by <a href="https://www.freepik.com" title="Freepik">Freepik</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a></div>

<div>Icons made by <a href="https://www.freepik.com" title="Freepik">Freepik</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a></div>

<div>Icons made by <a href="https://www.flaticon.com/authors/mavadee" title="mavadee">mavadee</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a></div>

### CODE REVIEW

File: app/src/main/res/layout/activity_single_weather_report.xml [hunk 1]
```diff
@@ -1,106 +1,458 @@
...
+                android:layout_weight="0"
+                app:srcCompat="@drawable/ic_download" />
+
+            <Space
+                android:layout_width="match_parent"
+                android:layout_height="42dp"
+                android:layout_weight="0" />
+
+            <TextView
+                android:id="@+id/textView4"
```
Comment (JackMcGill) [inline app/src/main/res/layout/activity_single_weather_report.xml]: Hi @CalebWebsterJCU, the screenshots you sent of the layout look spot on, good work. 

As I can’t compile and run the project I can’t give you a full review, that will have to come from someone else who can run your code. From what I can see though, your XML view properties have non-descriptive names, so they will be hard to work with when they’re being accessed by the Java code. Check our [contributing](https://github.com/JackMcKill/CP3407-Project/blob/develop/Contributing.md) page to see more info on naming, but the names should be something more along these lines: 
```suggestion
                android:id="@+id/tv_maxTemp"
```

**Review by JackMcGill — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/activity_single_weather_report.xml (hunk 1 above)
Comment (JackMcGill) [reply app/src/main/res/layout/activity_single_weather_report.xml]: Apart from the property names, everything else looks really good that I can see. Nice work

**Comment by HarperThurlow** [comment]
I am getting a "AAPT: error: resource drawable/ic_barometer (aka com.cp3407.wildernessweather:drawable/ic_barometer) not found." error when trying to run your code. My guess is that something that you have used in your code is not saved in the actual project structure and locally on your computer (probably ic_barometer). This might not be the only one but is the only one I can see for now.

File: app/src/main/res/layout/activity_single_weather_report.xml (hunk 1 above)
Comment (calebwebster) [reply app/src/main/res/layout/activity_single_weather_report.xml]: Thanks, I will update the names.

**Comment by calebwebster** [comment]
@HarperThurlow I also noticed that the barometer image (thats what it is) was not contained in the PR. I'll see if I can sort it out.

**Comment by HarperThurlow** [comment]
Another thing I am noticing is you have very set in stone widths. This wont be good for if someone is using anothe device (either bigger or smaller). To fix this we normally use Percentages instead of exact dp measurements. 
Also what Jack has said above is super important as it makes better documentation too so we know what button/image/textview is what. Wouldn't hurt to add some small comments saying what it is within the object aswell.

**Comment by calebwebster** [comment]
I'm changing everything over to percentages, would you advise dividing the LinearLayouts into parts using weightSum, or treating it as one part, with the widgets occupying a percentage of that one part?

**Comment by HarperThurlow** [comment]
Not quiet sure what you mean. Wanna jump on a call on discord?

**Comment by calebwebster** [comment]
Ok I just learned that weightSum is the sum of the weights of all its widgets, I've set weightSum to 100 so that everything inside can be a percentage value for the layout_weight.

**Comment by calebwebster** [comment]
Finished converting to percentages.

**Review by JackMcGill — APPROVED** [review]

**Comment by JackMcGill** [comment]
Thanks for making those changes @CalebWebsterJCU, it's looking good. Before you merge this branch however, we should probably decide what we are going to do about the `AppBar` + the blue bar that's going on:  

Screen Shot 2021-10-06 at 7 16 07 pm. 

 Currently it looks a bit cluttered to me, having first the `AppBar` and then directly underneath it basically another one.
 
 I propose we get rid of the blue bar in the XML layout file, and instead program our `AppBar` itself to be blue and display the city name (rather than saying WildernessWeather), along with a left-facing chevron to work as a "back button" or "up button". What do you think @CalebWebsterJCU? 

There is an example of how to change the text displayed in the `AppBar` on the `SettingsActivity` file. You can very easily set the text (and I imagine colour as well) for it in the `onCreate` method.

**Review by HarperThurlow — APPROVED** [review]
Thanks for fixing those issues @CalebWebsterJCU. All looks good to me now 👍

**Pull request merged**